tests/test/tests.js

```diff
@@ -0,0 +1,411 @@
+const chaiAsPromised = require("chai-as-promised");
+
+/* Boilerplate start */
+var electronPath = path.join(
+  __dirname,
+  "../..",
+  "node_modules",
+  ".bin",
+  "electron"
+);
+
+if (process.platform === "win32") {
+  electronPath += ".cmd";
+}
+
+var appPath = path.join(__dirname, "../..");
+
+var app = new Application({
+  path: electronPath,
+  args: [appPath]
+});
+
+global.before(function() {
+  chai.should();
+  chai.use(chaiAsPromised);
+});
+/* Boilerplate End */
+
+describe("Terminal Class Check", () => {
+  //Checks a terminal can be created
+  it("Check no argument terminal", () => {
+    let terminal = Terminal.createTerminal("whoami");
+    return terminal.spawnfile === "whoami";
+  });
+
+  //Checks a terminal with a single argument can be created
+  it("Check single arguement terminal", () => {
+    let args = ["-option1"];
+    let terminal = Terminal.createTerminal("whoami", args);
+
+    if (terminal.spawnfile !== "whoami") return false;
+    for (let i = 0; i < args.length; i++) {
+      if (terminal.spawnargs[i] !== args[i]) return false;
+    }
+    return true;
+  });
+
+  //Checks a terminal with multiple arguements can be created
+  it("Check multi arguement terminal", () => {
+    let args = ["-option1", "-option2"];
+    let terminal = Terminal.createTerminal("whoami", args);
+
+    if (terminal.spawnfile !== "whoami") return false;
+    for (let i = 0; i < args.length; i++) {
+      if (terminal.spawnargs[i] !== args[i]) return false;
+    }
+    return true;
+  });
+
+  //Checks an AP specific command can be created
+  it("Check single arguement AP command", () => {
+    const AP_NAME = "AnaysisPrograms.exe";
+    let terminal = Terminal.createAPTerminal(["list"]);
+
+    if (process.platform === "win32") {
+      let command = terminal.spawnfile.substr(
+        terminal.spawnfile.length - AP_NAME.length,
+        AP_NAME.length
+      );
+      return (
+        command === AP_NAME &&
+        terminal.spawnargs.length === 1 &&
+        terminal.spawnargs[0] === "list"
+      );
+    } else {
+      let ap = terminal.spawnargs[0].substr(
+        terminal.spawnfile.length - AP_NAME.length,
+        AP_NAME.length
+      );
+      return (
+        terminal.spawnfile === "mono" &&
+        terminal.spawnargs.length === 2 &&
+        ap === AP_NAME &&
+        terminal.spawnargs[1] === "list"
+      );
+    }
+  });
+
+  //Checks an AP specific command with multiple arguements can be created
+  it("Check multi arguement AP command", () => {
+    const AP_NAME = "AnaysisPrograms.exe";
+    let args = ["audio2csv", "help"];
+    let terminal = Terminal.createAPTerminal(args);
+
+    if (process.platform === "win32") {
+      let command = terminal.spawnfile.substr(
+        terminal.spawnfile.length - AP_NAME.length,
+        AP_NAME.length
+      );
+      return (
+        command === AP_NAME &&
+        terminal.spawnargs.length === args.length &&
+        terminal.spawnargs[0] === args[0] &&
```

Comment:
I'd much rather see a string here (applies to all tests like this).

```
terminal.spawnargs.join(" ") === "audio2csv help" 
```

It is easier to read, and easier to understand the intention